homeassistant/components/bosch_shc/sensor.py

```diff
@@ -0,0 +1,401 @@
+        return f"{self._device.name} Energy"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.energyconsumption / 1000.0
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return ENERGY_KILO_WATT_HOUR
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return "mdi:gauge"
+
+
+class BatterySensor(SHCEntity):
+    """Representation of a SHC battery reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_battery"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Battery"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if (
+            self._device.batterylevel
+            == SHCBatteryDevice.BatteryLevelService.State.CRITICAL_LOW
+        ):
+            _LOGGER.warning("Battery state of device %s is critical low", self.name)
+            return 0
+
+        if (
+            self._device.batterylevel
+            == SHCBatteryDevice.BatteryLevelService.State.LOW_BATTERY
+        ):
+            _LOGGER.warning("Battery state of device %s is low", self.name)
+            return 20
+
+        if self._device.batterylevel == SHCBatteryDevice.BatteryLevelService.State.OK:
+            return 100
+
+        if (
+            self._device.batterylevel
+            == SHCBatteryDevice.BatteryLevelService.State.NOT_AVAILABLE
+        ):
+            _LOGGER.debug("Battery state of device %s is not available", self.name)
+
+        return None
+
+    @property
+    def device_class(self):
+        """Return the class of the sensor."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return PERCENTAGE
+
+
+class ValveTappetSensor(SHCEntity):
+    """Representation of a SHC valve tappet reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_valvetappet"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Valvetappet"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.position
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return "mdi:gauge"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of the sensor."""
+        return PERCENTAGE
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "valve_tappet_state": self._device.valvestate.name,
```

Comment:
What does this measurement represent? Can we make a separate sensor instead that is disabled by default?

Reply:
The state is an enum with values `VALVE_ADAPTION_SUCCESSFUL`, `VALVE_ADAPTION_IN_PROGRESS`, `RANGE_TOO_BIG`, `RUN_TO_START_POSITION`, `IN_START_POSITION`. They actually belong to the valve tappet position, so I would keep them as state attributes.

Reply:
Yes, good to keep them here.